lib/module-utils.js

```diff
@@ -0,0 +1,171 @@
+const eslint = require('eslint')
+const fs = require('fs')
+const minimatch = require('minimatch')
+const moduleVisitor = require('eslint-module-utils/moduleVisitor')
+const parse = require('eslint-module-utils/parse').default
+const resolve = require('eslint-module-utils/resolve').default
+
+exports.detectUsedModuleExports = function(files) {
+  return Promise.all(files.map(exports.load)).then(results => {
+    const imported = new Map()
+
+    for (const {path, ignore, exports} of results) {
+      if (ignore) {
+        continue
+      }
+
+      const usedExports = new Map()
+      imported.set(path, usedExports)
+      for (const identifier of exports) {
+        usedExports.set(identifier, false)
+      }
+    }
+
+    for (const {imports} of results) {
+      for (const [importedPath, importedIdentifiers] of imports) {
+        if (imported.has(importedPath)) {
+          for (const importedIdentifier of importedIdentifiers) {
+            if (importedIdentifier === '*') {
```

Comment:
Does this cover the detection that, provided that module `foo` exports `a` and `b`, the unused export gets detected here:

```js
import * as f from "foo"
f.a()
```

Reply:
It should, `"*"` was an internal sentinel I used to indicate `import * as f from "foo"` or `import "foo"`.